Updated Mapper for Gen3 Metadata Schema (First Pass, AZUL-5)

GPelayo opened the pull request:
Update the Azul indexer mappings for metadata exported from the UChicago Gen3 system for the TOPMed 107 open access data set.

Also changed transformer mappings to be fully qualified JSON paths, and changed the transformer implementation from using jsonpath to using jmespath.

Commit:
Change transformer to accept full qualified paths for identifying metadata fields. Replaced jsonpath_rw library with jmespath.

Commit:
Updated Mapping for Gen3 schema (first pass)

Commit:
Updated Mapping for Gen3 schema (/w full qualified path)

Commit:
Added file_type to mappings

mikebaumann commented on chalicelib/dcc/transformer.py:
```diff
@@ -123,15 +139,12 @@ def transform(self, data_file, metadata_files):
-                raise
-            return result_value
+                log.error("The '{}' field doesn't exist in selected metadata".format(field))
+                exit()
```
An indexing error should be raised (as it was before), and not cause an exit.

mikebaumann commented on chalicelib/dcc/transformer_mapping.json:
```diff
@@ -7,47 +7,28 @@
-        "dss_field": "center_name",
-        "index_field": "center_name"
+        "dss_field": "project_id",
+        "index_field": "core_metadata.project"
```
These values show be swapped (`core_metadata.project` is a `dss_field`, and vice-versa).

mikebaumann commented on chalicelib/dcc/transformer_mapping.json:
```diff
@@ -7,47 +7,28 @@
         "from_file_data": true,
         "dss_field": "size",
-        "index_field": "fileSize"
+        "index_field": "file_size"
```
This should be `fileSize` as it was before. Sorry I misspoke on this.

mikebaumann commented on chalicelib/dcc/transformer_mapping.json:
```diff
@@ -7,47 +7,28 @@
-      },
-      {
-        "source": "metadata.json",
+        "from_file_data": true,
```
Let's set `project`, `program` to the value of `core_metadata.project` as these values are expected to be there by the current `dcp-dashboard-service` configuration and requests to the dashboard service fail if they are not there. The project and program fields will like likely want to keep.

mikebaumann commented on chalicelib/dcc/transformer_mapping.json:
```diff
@@ -7,47 +7,28 @@
       },
       {
         "source": "metadata.json",
-        "index_field": "analysis_type",
```
Let's keep "analysis_type" because the dashboard service expects it to be there and I think we will likely be able to get values for this at some point in the future. Let's make it an empty string constant for now.

mikebaumann commented on chalicelib/dcc/transformer_mapping.json:
```diff
@@ -65,103 +46,85 @@
       },
       {
         "source": "metadata.json",
-        "index_field": "donor",
```
Let's keep "donor" because the dashboard service expects it to be there and I think we will likely be able to get values for this at some point in the future. Let's make it an empty string constant for now.

mikebaumann commented on chalicelib/dcc/transformer_mapping.json:
```diff
@@ -7,47 +7,28 @@
       },
       {
-        "source": "metadata.json",
-        "dss_field": "donor_uuid",
```
I expect `redwoodDonorUUID` will likely change to something else or go away at some point, yet for now it is needed for the dashboard service to work.  For the time being, let's use the constant mapping to set these to an empty string.

mikebaumann reviewed (changes requested):
Thanks for the great work on this, Geryl!
Almost there, I think. Please let me know if you have questions.

Commit:
Fixed Transformers Mappings for Boardwalk and Fixed Exception Handling

Commit:
Fixed Transformers Mappings for Boardwalk and Fixed Exception Handling

Commit:
Corrections to dcc/transformer_mappings.json

Corrected configuration for `analysis_type` and `project`
Removed duplicate mappings for `project` and `file_type`

Commit:
Updated requirements.txt to use hca 3.5.1

Commit:
Changed transformer_mapping.json to set "program" to "TOPMed"

Until the new loading mechanism is available, set the "program" field to
the constant value "TOPMed". When the new loading mechanism is available,
the "program" setting should use the value provided in the metadata.
